app/models/labels/plate_label_cellaca_qc.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Plate label class to print off the QC labels for the cellaca process
+# This label template can generate up to 4 qc labels depending on well
+# occupancy.
+class Labels::PlateLabelCellacaQc < Labels::PlateLabelBase
+  COLS_PER_PAGE = 3
+
+  def attributes
+    super.merge(barcode: labware.barcode.human)
+  end
+
+  def qc_attributes
+    max_qc_plates.times.filter_map do |index|
+      # Dividing a column by three maps it to its page
+      next if occupied_columns.none? { |col| col / COLS_PER_PAGE == index }
+
+      qc_label(index)
+    end
+  end
+
+  private
+
+  def qc_label(index)
```

Comment:
This is better than what I had before :)
One comment, does it print them out in reverse order, i.e. 4,3,2,1 rather than 1,2,3,4?
Easy for the user if we order reversed here so when you print them they are in right order top to bottom on the physical labels (as it prints labels upside down essentially).

Reply:
Ahh, no, I'd need to add a reverse on the end for that.